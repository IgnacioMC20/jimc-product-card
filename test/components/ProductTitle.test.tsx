import * as React from 'react';
import TestRenderer from 'react-test-renderer';
import { ProductTitle, ProductCart } from '../../src/components';
import { product } from '../data/products';

describe('ProductTitle', () => {

    test('debe mostrar el componente correctamente con el titulo personalizado', () => {
        const wrapper = TestRenderer.create(
            <ProductTitle title='CustomProduct' />
        )
        expect(wrapper.toJSON()).toMatchSnapshot()
    })

    test('debe mostrar el componente con el nombre del prodcuto', () => {
        const wrapper = TestRenderer.create(
            <ProductCart product={product}>
                {() => (
                    <ProductTitle />
                )}
            </ProductCart>
        )
        expect(wrapper.toJSON()).toMatchSnapshot()
    })
})