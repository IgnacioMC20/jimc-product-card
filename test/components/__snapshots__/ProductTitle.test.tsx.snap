// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle debe mostrar el componente con el nombre del prodcuto 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee Mug - Card no img
  </span>
</div>
`;

exports[`ProductTitle debe mostrar el componente correctamente con el titulo personalizado 1`] = `
<span
  className="productDescription undefined"
>
  CustomProduct
</span>
`;
